src/DGtal/geometry/surfaces/estimation/PlaneProbingTetrahedronEstimator.ih

```diff
@@ -0,0 +1,414 @@
+template < typename TPredicate, DGtal::ProbingMode mode >
+inline
+DGtal::PlaneProbingTetrahedronEstimator<TPredicate, mode>::
+PlaneProbingTetrahedronEstimator (Point const& aPoint, Triangle const& aM, Predicate const& aPredicate)
+    : myM(aM), myPredicate(aPredicate), myS(aM[0] + aM[1] + aM[2]), myQ(aPoint + myS)
+{
+    myNeighborhood = DGtal::detail::PlaneProbingNeighborhoodSelector<TPredicate, mode>::select(myPredicate, myQ, myM);
+
+    if (! isValid())
+    {
+        throw std::runtime_error("Invalid frame passed in PlaneProbingTetrahedronEstimator constructor");
+    }
+}
+
+// ------------------------------------------------------------------------
+template < typename TPredicate, DGtal::ProbingMode mode >
+inline
+DGtal::PlaneProbingTetrahedronEstimator<TPredicate, mode>::
+~PlaneProbingTetrahedronEstimator ()
+{
+    delete myNeighborhood;
+    myNeighborhood = nullptr;
+}
+
+///////////////////////////////////////////////////////////////////////////////
+// ----------------------- Plane probing services ------------------------------
+
+// ------------------------------------------------------------------------
+template < typename TPredicate, DGtal::ProbingMode mode >
+inline
+typename DGtal::PlaneProbingTetrahedronEstimator<TPredicate, mode>::Point const&
+DGtal::PlaneProbingTetrahedronEstimator<TPredicate, mode>::m (int aIndex) const
+{
+    assert(aIndex == 0 || aIndex == 1 || aIndex == 2);
+    return myM[aIndex];
+}
+
+// ------------------------------------------------------------------------
+template < typename TPredicate, DGtal::ProbingMode mode >
+inline
+typename DGtal::PlaneProbingTetrahedronEstimator<TPredicate, mode>::Point const&
+DGtal::PlaneProbingTetrahedronEstimator<TPredicate, mode>::q () const
+{
+    return myQ;
+}
+
+// ------------------------------------------------------------------------
+template < typename TPredicate, DGtal::ProbingMode mode >
+inline
+typename DGtal::PlaneProbingTetrahedronEstimator<TPredicate, mode>::Point
+DGtal::PlaneProbingTetrahedronEstimator<TPredicate, mode>::getOrigin () const
+{
+    return myQ - (myM[0] + myM[1] + myM[2]);
+}
+
+// ------------------------------------------------------------------------
+template < typename TPredicate, DGtal::ProbingMode mode >
+inline
+typename DGtal::PlaneProbingTetrahedronEstimator<TPredicate, mode>::Triangle
+DGtal::PlaneProbingTetrahedronEstimator<TPredicate, mode>::vertices () const
+{
+    return { myQ - myM[0], myQ - myM[1], myQ - myM[2] };
+}
+
+// ------------------------------------------------------------------------
+template < typename TPredicate, DGtal::ProbingMode mode >
+inline
+std::pair<typename DGtal::PlaneProbingTetrahedronEstimator<TPredicate, mode>::Point,
+          typename DGtal::PlaneProbingTetrahedronEstimator<TPredicate, mode>::Point>
+DGtal::PlaneProbingTetrahedronEstimator<TPredicate, mode>::getBasis () const
+{
+    Point u = myM[1] - myM[0],
```

Comment:
I think it is no harm to use a "using DGtal::detail" within this method.

Reply:
You're right, it's cleaner that way.